src/core/tracking/skb_tracking.rs

```diff
@@ -220,7 +220,13 @@ pub(crate) fn init_tracking(
 
         while run_state.running() {
             // Let's run every SKB_TRACKING_GC_INTERVAL seconds.
-            thread::sleep(Duration::from_secs(SKB_TRACKING_GC_INTERVAL));
+            for _ in 0..SKB_TRACKING_GC_INTERVAL {
+                thread::sleep(Duration::from_secs(1));
+                if !run_state.running() {
+                    break;
```

Comment:
Mostly cosmetic (and optional), but I guess a return here makes it more obvious (it is fairly obvious already, tbf :)
Also (optional :), if moving the running() check before sleeping on top of it, allows to get rid of the "while" condition and just loop {} which looks a bit redundant. Not exactly the same, but should be ok.
But if you prefer to keep it so, lgtm.

Reply:
ack, looks good